### DISCUSSION TITLE
How to create an Error for any StatusCode

### DISCUSSION THREAD

**Discussion by jayvdb** [actix/actix-web#2980] [Q&A] [answered]
If I want an `actix_web::Error`, I know I can do `std::io::Error::new(std::io::ErrorKind::PermissionDenied, "not possible").into()` (403), and similar for `std::io::ErrorKind::NotFound` (404) , but how do I do that for any value in `StatusCode` ?  It seems there is not `impl ResponseError for StatusCode {.. }`.

If there isnt a way yet, and adding a `impl ResponseError for StatusCode {.. }` is appropriate, I'd be happy to add it.

**Comment by robjtede** [comment, marked as answer]
Best thing to do is use [`InternalError`](https://docs.rs/actix-web/latest/actix_web/error/struct.InternalError.html#method.new).

`ResponseError` is designed to require some error context. So such an `impl ResponseError for StatusCode` is not acceptable.

**Comment by bublikOff** [comment]
Got following  error generation code ... 
`Error::from(InternalError::new("Authentication Timeout", http::StatusCode ::from_u16(419).unwrap()))`

it returns code 419 ... but reason has following value `unknown status code`
How to set reason ?

**Reply by robjtede** [reply]
You'll have to create the `HttpResponse` yourself and use [this method](https://docs.rs/actix-web/latest/actix_web/struct.HttpResponseBuilder.html#method.reason).

**Reply by bublikOff** [reply]
Yes, I can 
`HttpResponse::build(HttpStatusCode::from_u16(419).unwrap()).reason("Authentication Timeout").body("Authentication Timeout")`
But how to convert it to Error then ? and return like I do with error 401
`ready(Err(actix_web::error::ErrorUnauthorized("Unauthorized")));`

**Reply by robjtede** [reply]
https://docs.rs/actix-web/latest/actix_web/error/struct.InternalError.html#method.from_response

**Reply by bublikOff** [reply]
have missed that... thx a lot

**Comment by jayvdb** [comment]
Thanks for the suggestions. We switched to using thiserror and https://github.com/enzious/actix-web-thiserror , which meant we no longer needed to instantiate the `Err(..)` objects manually.
